stingray/varenergyspectrum.py

```diff
@@ -186,18 +186,23 @@ def _spectrum_function(self):
         for i, eint in enumerate(self.energy_intervals):
             base_lc, ref_lc = self._construct_lightcurves(eint,
                                                           exclude=False)
-            xspect = AveragedCrossspectrum(base_lc, ref_lc,
-                                           segment_size=self.segment_size,
-                                           norm='frac')
-            good = (xspect.freq >= self.freq_interval[0]) & \
-                   (xspect.freq < self.freq_interval[1])
-            rms_spec[i] = np.sqrt(np.sum(xspect.power[good]*xspect.df))
-
-            # Root squared sum of errors of the spectrum
-            root_sq_err_sum = np.sqrt(np.sum(xspect.power[good]**2))*xspect.df
-            # But the rms is the squared root. So,
-            # Error propagation
-            rms_spec_err[i] = 1 / (2 * rms_spec[i]) * root_sq_err_sum
+            try:
+                xspect = AveragedCrossspectrum(base_lc, ref_lc,
+                                               segment_size=self.segment_size,
+                                               norm='frac')
+            except AssertionError as e:
+                # Avoid "Mean count rate is <= 0. Something went wrong" assertion.
+                simon("AssertionError: " + str(e))
```

Comment:
so what happens when that assertion error is thrown? It's caught and printed with a SIMON warning, but won't it still break the method, because `rms_spec` and `rms_spec_err` won't be defined?